library-best-practices/glossary.md

```diff
@@ -3,14 +3,20 @@ Java Dependency Glossary
 
 ### Types of conflicts and compatibility
 
+- **Linkage error**: an error when a Java class in a classpath has reference to
+  another class file for a class, field or method, and the reference can not be
+  satisfied with the available classes in the classpath.
+  A linkage error is caused by a reference to missing classes or a linkage conflict.
+  Linkage errors detected at runtime manifest as `ReflectiveOperationException`,
+  `NoClassDefFoundError`, `NoSuchFieldException`, `MethodNotFoundException`,
+  `LinkageError`, or other related exceptions.
+  - Sub-type: **Linkage conflict**
+
 - **Linkage conflict**: The signature, return type, modifiers, or throws
```

Comment:
I think we should indent "linkage conflict" since it is a subtype of "linkage error", and also add a definition for a term referring to missing things.